app/controllers/pensions/PensionsSummaryController.scala

```diff
@@ -43,15 +45,22 @@ class PensionsSummaryController @Inject() (implicit
     extends FrontendController(mcc)
     with I18nSupport {
   def show(taxYear: Int): Action[AnyContent] = (authAction andThen taxYearAction(taxYear)).async { implicit request =>
-    pensionSessionService.getAndHandle(taxYear, request.user) { (pensionsUserData, prior) =>
-      if (pensionsUserData.exists(!_.pensions.isEmpty)) {
-        Future.successful(Ok(pensionsSummaryView(taxYear, pensionsUserData.map(_.pensions), prior)))
-      } else {
-        /* -- initialise pensions session date either with prior data or an empty model -- */
-        val cya = prior.fold(PensionsCYAModel.emptyModels)(pr => AllPensionsData.generateCyaFromPrior(pr))
-        pensionSessionService.createOrUpdateSessionData(request.user, cya, taxYear, isPriorSubmission = prior.isDefined)(
-          errorHandler.handleError(INTERNAL_SERVER_ERROR))(Ok(pensionsSummaryView(taxYear, Some(cya), prior)))
-      }
+    pensionSessionService.getAndHandle(taxYear, request.user) { (pensionsUserData: Option[PensionsUserData], priorData: Option[AllPensionsData]) =>
```

Comment:
We could rename `pensionsUserData` to `sessionData` to be completely clear?

Reply:
done